gk/main.c

```diff
@@ -686,6 +794,352 @@ gk_conf_put(struct gk_config *gk_conf)
 }
 
 static int
+setup_gk_lpm(struct gk_config *gk_conf, unsigned int socket_id)
+{
+	int ret;
+	struct rte_lpm_config ipv4_lpm_config;
+	struct rte_lpm6_config ipv6_lpm_config;
+	struct gk_lpm *ltbl = &gk_conf->lpm_tbl;
+
+	ipv4_lpm_config.max_rules = gk_conf->max_num_ipv4_rules;
+	ipv4_lpm_config.number_tbl8s = gk_conf->num_ipv4_tbl8s;
+	ipv6_lpm_config.max_rules = gk_conf->max_num_ipv6_rules;
+	ipv6_lpm_config.number_tbl8s = gk_conf->num_ipv6_tbl8s;
+
+	ltbl->lpm = init_ipv4_lpm("gk", &ipv4_lpm_config, socket_id, 0, 0);
```

Comment:
In this case, there must be added a comment to the function stating that the last two parameters are only used to differentiate instances. This is important because you can only set a lcode ID with an arbitrary value due to this information. In addition, you should add a comment to the call `init_ipv4_lpm()` above, and another to call `init_ipv6_lpm()` below stating the reason you're passing the zeros.

The comments are required because the code is unusual, but the reasoning present in the comments justifies it.